src/plugins/intel_cpu/src/nodes/istft.cpp

```diff
@@ -0,0 +1,111 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "istft.h"
+
+#include "nodes/common/cpu_memcpy.h"
+#include "openvino/core/type.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/istft.hpp"
+#include "openvino/reference/istft.hpp"
+
+namespace ov {
+namespace intel_cpu {
+namespace node {
+
+bool ISTFT::isSupportedOperation(const std::shared_ptr<const ov::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        if (op->get_type_info() != op::v16::ISTFT::get_type_info_static()) {
+            errorMessage = "Only ISTFT operation from the opset16 is supported by the CPU plugin.";
+            return false;
+        }
+    } catch (...) {
+        return false;
+    }
+    return true;
+}
+
+ISTFT::ISTFT(const std::shared_ptr<ov::Node>& op, const GraphContext::CPtr& context)
+    : Node(op, context, NgraphShapeInferFactory(op)) {
+    std::string errorMessage;
+    if (!isSupportedOperation(op, errorMessage)) {
+        OPENVINO_THROW_NOT_IMPLEMENTED(errorMessage);
+    }
+
+    const auto istft_op = as_type_ptr<op::v16::ISTFT>(op);
+
+    m_is_frame_size_const = is_type<op::v0::Constant>(istft_op->get_input_node_ptr(FRAME_SIZE_IDX));
+    m_is_frame_step_const = is_type<op::v0::Constant>(istft_op->get_input_node_ptr(FRAME_STEP_IDX));
+    if (istft_op->get_input_size() == 5) {
```

Comment:
Since input size is checked multiple times, consider adding additional bool variable that signal_length input is available

Reply:
Agree, applied.